### PR TITLE
Updated tests and 404 fallthrough in certain routes

### PR DESCRIPTION
**Pull request by CarlosFdez**
EDITED:

The dashboard endpoint and a few more bundle related endpoints now call next() instead of immediately 404ing. This allows extensions to handle those cases and create "fake files" instead of silently failing. One such instance is react-dev-middleware and another is potentially HMR.

Tests were also updated to improve consistency (they were failing on my machine and I needed to pass the coverage check) and a new test was added for dashboard 404.

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Yup, added sounds as a last minute one and I thought I had tested it but I guess I didn't. Don't merge this in just yet.... :V

**Comment by CarlosFdez** [comment]
Confirming on retesting that the test changes are placebo. Seems to be an intermittent issue related to computer performance. My guess is that something accessing the DOM or something is jumping the gun.

**Comment by CarlosFdez** [comment]
Seems to have improved, but tests will occasionally still fail locally. Current intermittent failures seem due to opening too many chromium windows simultaneously, or probably tests that employ hover when tabs are created in non-headless mode. Fixing those are out of the scope of this PR.

File: test/helpers/browser.js
Comment (Lange) [inline test/helpers/browser.js]: Can you explain why this was removed?

File: test/registration.js
Comment (Lange) [inline test/registration.js]: Can you explain why this was removed?

File: test/helpers/browser.js
Comment (CarlosFdez) [reply test/helpers/browser.js:102]: Short answer: It does nothing except cause unnecessary test failures

Long answer: Few reasons
1) What evaluate does is run some code in an injected script then return the result. This did a boolean check and then...do nothing with the result.
2) The other init functions called waitForFunction instead of evaluate, so I figured it was a typo. But then swapping it out caused test failures to happen all the time.
3) The test that occasionally failed because of that line is `singleInstance - should redirect to busy.html when the instance is already taken`. Basically injecting code (using evaluate) caused failures because it was redirecting, because of a race condition. waitForFunction similarly failed because it was redirecting, so that if condition was no longer true (and its not supposed to be true if it redirects).

There might be a better alternative than removing (a third option), but I'm not familiar enough to make that call.

File: test/registration.js
Comment (CarlosFdez) [reply test/registration.js:162]: This one I'm less sure about than the other one. This line occasionally caused failures with `registration » refresh individual instance` with the message `Node is either not visible or not an HTMLElement`. It didn't seem instrumental to the test (the goal is to click the element, it looked like hover was there because it was trying to simulate a human?), and the other instance where the reload button was clicked did not have the hover.

Basically removed for consistency reasons and because of the false negative I received when it existed.

File: test/helpers/browser.js
Comment (Lange) [reply test/helpers/browser.js]: Whoops, you're right. That code does nothing. I wonder if I meant for that to be a recurring check that waited for `window.singleInstanceApi` to exist before resolving the promise.

File: test/registration.js
Comment (Lange) [reply test/registration.js]: Kk, that sounds reasonable. I admit I have no idea why it was there to begin with.

File: test/helpers/browser.js
Comment (CarlosFdez) [reply test/helpers/browser.js:102]: Probably. It definitely looked like it was meant to be like the waitForFunctions above. I was going to make it consistent but then that redirect caused hella craziness. Felt sad not to see an assurance that it navigated correctly, but page.goto does a waitUntil: load anyways so its most likely fine.

Funny enough, I nearly commited a bunch of `, { waitUntil: 'load' }` before I realized that load is the default. Puppeteer is so much better than selenium.

**Comment by Lange** [comment]
Alright, I think I'm good with this. Just want to get a second opinion from @Hoishin before merging.

File: test/registration.js
Comment (Hoishin) [reply test/registration.js:162]: I think I added the line because sometimes I couldn't click the desired element and some stackoverflow suggested hovering before click, if I remember correctly.
If the tests work fine now it should be OK.

**Comment by Hoishin** [comment]
I think I've seen this discussed, but does it return 404 if you do `next()` and eventually no router sends response?

**Review by Hoishin — APPROVED** [review]
Thanks for clarification. LGTM

**Pull request merged**
